File: LeetCodeDSA/122.c
#include <stdio.h>

int maxProfit(int* prices, int pricesSize) {
    if(pricesSize == 1){
        return 0;
    }

    int total = 0; 
    int profit = 0; 
    int i = 0; 
    int j = 1; 
    while(i < pricesSize && j < pricesSize){
        while(j < pricesSize && prices[j] - prices[i] > profit){
            profit = prices[j] - prices[i]; 
            j++;
        }
        total += profit; 
        i = j; 
        j++; 
        profit = 0;
    }
    return total;
}
// huong di khac cua bai nay 
int maxProfit2(int* prices, int pricesSize){
    int total = 0; 
    for(int i = 1; i < pricesSize; i++){
        if(prices[i] > prices[i - 1]){ 
            total += prices[i] - prices[i - 1];
        }
    }
    return total;
}

int main(){
    int pricesSize = 6; 
    int prices[] = {3, 2, 6, 5, 0, 3};
    printf("%d", maxProfit(prices, pricesSize));
    return 0;
}

File: LeetCodeDSA/122.py
def maxProfit(prices):
    if len(prices) == 1:
        return 0
    total = 0
    profit = 0

    i = 0
    j = 1
    while i < len(prices) and j < len(prices): 
        while j < len(prices) and prices[j] - prices[i] > profit:
            profit = (prices[j] - prices[i] )
            j += 1
        total += profit
        i = j 
        j += 1
        profit = 0
    return total

def maxProfit2(prices): 
    profit = 0 
    i = 0 
    j = 1
    while i < len(prices) and j < len(prices): 
        if prices[j] - prices[i] > 0: 
            profit += prices[j] - prices[i]
        i += 1
        j += 1

    return profit

print(maxProfit(prices = [7,1,5,3,6,4]))
print(maxProfit2(prices = [7,1,5,3,6,4]))